src/lib.rs

```diff
@@ -148,10 +148,24 @@ pub trait AgnostikExecutor {
         T: Send + 'static;
 
     /// Blocks until the future has finished.
-    fn block_on<F, T>(&self, future: F) -> T
+    fn block_on<F>(&self, future: F) -> F::Output
     where
-        F: Future<Output = T> + Send + 'static,
-        T: Send + 'static;
+        F: Future + Send + 'static,
+        F::Output: Send + 'static;
+}
+
+/// This trait represents an executor that can spawn local futures.
```

Comment:
that is capable of spawning futures onto the same thread. that's probably what local means in most places.

Reply:
Is this meant to be a requested change?

Reply:
yes, it was.